py/README.md

```diff
@@ -1,18 +1,103 @@
-# Python scripts
+# BitBox python scripts
+
+This directory contains scripts to talk to the BitBox device directly via the command line
+ (e.g. `send_message.py`, `load_firmware.py`).
+
+## Setup
+
+These instructions use Python 3, pip, and venv (a lightweight "virtual environment").
+Inside the virtual environment, `python` and `pip` refer to Python 3.
+
+All commands below assume you are in the `py/bitbox02` directory.
+
+### Requirements
+
+- python
+- pip >= 25
 
-**Important: pip >= 25 is required.**  
 Editable installs (`pip install -e …`) are only supported with pip 25 or newer.  
 Older pip versions will fail due to changes in how editable installs are handled (PEP 660).
 
-To use the scripts (e.g. `send_message.py`, `load_firmware.py`):
+### Installing dependencies
+
+Install the required Python dependencies listed in `requirements.txt`:
+
 ```bash
-pip3 install py/bitbox02
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
+```
+
+## Activate the virtual environment
+
+If you open a new shell, remember to re-activate the virtual environment,
+to use the scripts and communicate with the BitBox.
+
+```
+source .venv/bin/activate
 ```
 
-To work on the library/scripts and install in editable mode:
+You can deactivate a virtual environment by typing `deactivate` in your shell.
+
+
+## Communicate with the BitBox
+
+This assumes that the firmware was installed on the BitBox device. To flash the
+firmware manually read the next section or install with the official BitBoxApp.
+
+Connect your BitBox and "tap this side".
+
+List and execute all available commands by running:
+
+```bash
+python ./send_message.py
+```
+
+This command will list what commands are currently possible, depending on which
+mode the device currently is, i.e "Bootloader mode" accepts different commands.
+From here you can execute any command the BitBox accepts.
+
+```
+What would you like to do?
+- (1) List device info
+- (2) Change device name
+- (3) Get root fingerprint
+- (4) Retrieve zpub of first account
+- (5) Retrieve multiple xpubs
+- (6) …
+```
+
+When connecting the first time to an initialized but unpaired BitBox, the device
+will prompt to unlock and continue to compare and confirm the Noise pairing key.
+This is a one-time action.
+
+
+## Flash the firmware.bin
+
+To install / flash the firmware binary the device must be in "Bootloader mode",
+then the following command will flash the firmware.bin onto the BitBox.
+This assumes you have a `./firmware.bin`.
```

Comment:
Mention that this is for unsigned firmware and a only works for dev-devices. For prod devices, the binary has to be signed and `--debug` needs to be skipped.

Reply:
That was a missing piece, thanks. I think it makes sense to mention the simulator here for contributors that don't have a dev-device.